pkg/apis/serving/v1beta1/inference_service_validation.go

```diff
@@ -141,6 +146,40 @@ func validateInferenceService(isvc *InferenceService) (admission.Warnings, error
 	return allWarnings, nil
 }
 
+// validateMultiNodeVariables validates one of the following: tensor-parallel-size, pipeline-parallel-size, or WorkerSpec.Size
+func validateMultiNodeVariables(isvc *InferenceService) error {
+	if isvc.Spec.Predictor.Model != nil {
+		if envPipelineParallelSize, exists := utils.GetEnvVarValue(isvc.Spec.Predictor.Model.PredictorExtensionSpec.Container.Env, constants.PipelineParallelSizeEnvName); exists {
+			// pipelineParallelSize should be bigger than 1 (head + worker)
+			if intPipelineParallelSize, err := strconv.Atoi(envPipelineParallelSize); err == nil {
+				if intPipelineParallelSize < 2 {
+					return fmt.Errorf(InvalidPipelineParallelSizeValueError, isvc.Name, envPipelineParallelSize)
+				}
+			} else {
+				return fmt.Errorf(InvalidParallelSizeValueError, isvc.Name, envPipelineParallelSize)
```

Comment:
Maybe, take into consideration that the user may be using a single-node runtime, and just by coincidence there are environment variables named the same as the ones validated here. In that case, the validation should not happen.

Reply:
I wish there was a way to determine whether the runtime is single-node or multi-node using only the InferenceService custom resource (CR), but I'm not sure if such a method exists. I believe that validation should be primarily checked using the InferenceService CR. Therefore, I plan to check the values that need to be adhered to when using parallelism. Specifically, tensor-parallel-size should be greater than 0, and pipeline-parallel-size should not be used in a single-node.

Reply:
I used workerSpec in isvc to enable multi-node